app/components/util/PhraseUtil.js

```diff
@@ -6,16 +6,35 @@
    *
    * @ngInject
    */
-  function PhraseUtil(TransStatusService) {
+  function PhraseUtil(TransStatusService, _) {
 
     function getSaveButtonStatus(phrase) {
-      if (phrase.newTranslation === '') {
-        return TransStatusService.getStatusInfo('untranslated');
+      if (phrase.plural) {
+        var newTranslations = phrase.newTranslations,
+            translations = phrase.translations;
+        if (_.isEmpty(_.compact(newTranslations))) {
+          return TransStatusService.getStatusInfo('untranslated');
+        }
+        else if (_.compact(newTranslations).length !== newTranslations.length) {
+          return TransStatusService.getStatusInfo('needswork');
+        }
+        else if (!_.isEmpty(_.difference(newTranslations, translations))) {
```

Comment:
This has a logical error: difference will be empty going from `[ "foo", "bar"]` to `[ "bar", "foo" ]`. There is another place that this comparison is being made (commented earlier with a lodash approach to it), so I think we need a function that will tell if 2 arrays of strings are the same. With a dedicated function like `haveTranslationsChanged(original, current)` we will ensure we don't hit this pitfall.